src/topology.md

```diff
@@ -27,7 +41,12 @@ intermediate mixnodes should be uniformly randomly chosen, subject to the follow
 - No mixnode should appear in a route more than once, unless this is unavoidable. For example, when
   a non-mixnode is generating a loop route, if there is only a single connected gateway mixnode,
   the gateway must necessarily appear twice.
+  // TODO in this case I would find it better to return an error on route generation and put back
+  // the mixnet in a synching state.
```

Comment:
I didn't think reusing a gateway node was a particularly big deal. Allowing it means that non-mixnodes only need to successfully connect to a single mixnode for things to work.

Reply:
I mean if there is not enough mixnode to do a N hop route, then the N hop security asumption does not hold, it become N-(number of more than one peer access) security, but when sending the query we only did accept sending for a N security, it would need a message like "not enough node in network, do you still want to send your message Y/N"?

I mean my issue here is "unless this is unavoidable" part of the sentence.

Reply:
The runtime enforces a minimum bound on the number of mixnodes; if there are too few registered, the get-mixnodes function will simply return an error. There are only really two cases where you should end up with a route containing the same mixnode twice:

- If you are a mixnode and you're generating a loop route (obviously you will be in the route twice).
- If you are not a mixnode, you're generating a loop route, and there is only one connected gateway mixnode (this is the example given, the gateway mixnode needs to appear twice).

I figured the simplest way to word this was "unless this is unavoidable". But maybe this is misleading.

Reply:
I totally misunderstood the sentence here, I like the detail explanation of case better.
"except for cover loop where first mixnet node is the same as last mixnet node"
(when reading unavoidable I did think of a contextual temporary reason)

> If you are not a mixnode, you're generating a loop route, and there is only one connected gateway mixnode (this is the example given, the gateway mixnode needs to appear twice).

When reading I was thinking of route as either the query route or the surb route, so this case is not that relevant to me.
Actually do we consider that surb should not use mixnode from the query route (I think not)?